spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/annotation/SqsListenerAnnotationBeanPostProcessor.java

```diff
@@ -66,9 +70,15 @@ protected String getMessageListenerContainerRegistryBeanName() {
 	}
 
 	@Override
-	protected Collection<HandlerMethodArgumentResolver> createAdditionalArgumentResolvers() {
-		return Arrays.asList(new VisibilityHandlerMethodArgumentResolver(SqsHeaders.SQS_VISIBILITY_TIMEOUT_HEADER),
+	protected Collection<HandlerMethodArgumentResolver> createAdditionalArgumentResolvers(
+			MessageConverter messageConverter, ObjectMapper objectMapper) {
+		var additionalArgumentResolvers = Arrays.asList(new VisibilityHandlerMethodArgumentResolver(SqsHeaders.SQS_VISIBILITY_TIMEOUT_HEADER),
 				new SqsMessageMethodArgumentResolver(), new QueueAttributesMethodArgumentResolver());
+		if (messageConverter != null && objectMapper != null) {
```

Comment:
here is where I am handling the empty arguments, though maybe there is a better approach than this? I don't like it too much

Reply:
Maybe we can invert this - we can leave the old method as it was, and in the new method we call get the list from it and add the new resolver.

Wdyt?